src/App.tsx

```diff
@@ -1,19 +1,68 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { FilterBy } from './types/FilterBy';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+  const [hasError, setHasError] = useState(false);
+  const [todoTitle, setTodoTitle] = useState('');
+
+  const getTodosFromServer = async () => {
+    if (user) {
+      try {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      } catch {
+        setHasError(true);
+
+        setTimeout(() => {
+          setHasError(false);
+        }, 3000);
+      }
+    }
+  };
+
+  const deleteErrors = () => {
+    setHasError(false);
+  };
+
+  const visibleTodos = todos.filter(todo => {
+    switch (filterBy) {
+      case FilterBy.Completed:
+        return todo.completed === true;
+      case FilterBy.Active:
+        return todo.completed === false;
+      case FilterBy.All:
+      default:
+        return todo;
+    }
+  });
```

Comment:
It's better to keep visibleTodos in the state than a separate variable in the component

Reply:
I don't understand why do that here.